docs/install/Knative-with-any-k8s.md

```diff
@@ -1,89 +1,474 @@
+   Save this for configuring DNS below.
 
-1. To install Knative, first install the CRDs by running the `kubectl apply`
-   command once with the `-l knative.dev/crd-install=true` flag. This prevents
-   race conditions during the install, which cause intermittent errors:
+{{< /tab >}}
+
+{{% tab name="Kourier" %}}
+The following commands install Kourier and enables its Knative integration:
+
+1. Install the Knative Kourier controller:
 
    ```bash
-   kubectl apply --selector knative.dev/crd-install=true \
-   --filename https://github.com/knative/serving/releases/download/{{< version >}}/serving.yaml \
-   --filename https://github.com/knative/eventing/releases/download/{{< version >}}/eventing.yaml \
-   --filename https://github.com/knative/serving/releases/download/{{< version >}}/monitoring.yaml
+   kubectl apply --filename https://raw.githubusercontent.com/knative/serving/{{< version >}}/third_party/kourier-latest/kourier.yaml
    ```
 
-1. To complete the install of Knative and its dependencies, run the
-   `kubectl apply` command again, this time without the `--selector` flag, to
-   complete the install of Knative and its dependencies:
+2. To configure Knative Serving to use Kourier by default:
+
+    ```bash
+    kubectl patch configmap/config-network \
+      --namespace knative-serving \
+      --type merge \
+      --patch '{"data":{"ingress.class":"kourier.ingress.networking.knative.dev"}}'
+    ```
+
+3. Fetch the External IP or CNAME
 
    ```bash
-   kubectl apply --filename https://github.com/knative/serving/releases/download/{{< version >}}/serving.yaml \
-   --filename https://github.com/knative/eventing/releases/download/{{< version >}}/eventing.yaml \
-   --filename https://github.com/knative/serving/releases/download/{{< version >}}/monitoring.yaml
+   kubectl --namespace kourier-system get service kourier
    ```
 
+   Save this for configuring DNS below.
+
+{{< /tab >}}
+{{< /tabs >}}
+
+1. Configure DNS
+
+   <!-- This indentation is important for things to render properly. -->
+   {{< tabs name="serving_dns" >}}
+{{% tab name="Magic DNS (xip.io)" %}}
+We ship a simple Kubernetes Job called "default domain" that will (see caveats) configure Knative Serving to use <a href="http://xip.io">xip.io</a> as the default DNS suffix.
+
+```bash
+kubectl apply --filename https://github.com/knative/serving/releases/download/{{< version >}}/serving-default-domain.yaml
+```
+
+**Caveat**: This will only work if the cluster LoadBalancer service exposes an IPv4 address, so it will not work with IPv6 clusters, AWS, or local setups like Minikube.  For these, see "Real DNS".
+{{< /tab >}}
+
+
+{{% tab name="Real DNS" %}}
+To configure DNS for Knative, take the External IP or CNAME from setting up networking, and configure it with your DNS provider as follows:
+
+- If the networking layer produced an External IP address, then configure a wildcard A record for the domain:
```

Comment:
Is "a wildcard A record" correct, or should it be "a wildcard record"?

Reply:
I made it `A` to clarify.